<launch>
  <arg name="right_electric_gripper" default="true"/>
  <arg name="left_electric_gripper" default="true"/>
  <arg name="left_tip_name" default="left_gripper"/>
  
  <arg name="right_tip_name" default="right_gripper"/>
  <param name="robot_description"
      command="$(find xacro)/xacro.py --inorder $(find baxter_description)/urdf/baxter.urdf.xacro"/>
  <param name="robot_description_semantic"
      command="$(find xacro)/xacro.py --inorder $(find alternative_baxter_moveit_config)/config/baxter.srdf
          left_electric_gripper:=$(arg left_electric_gripper)
          right_electric_gripper:=$(arg right_electric_gripper)
          left_tip_name:=$(arg left_tip_name)
          right_tip_name:=$(arg right_tip_name)"/>

  <!--group ns="robot_description_planning">
    <rosparam command="load" file="$(find baxter_moveit_config)/config/joint_limits.yaml"/>
  </group-->

  <!--param name="robot_description"
      command="$(find xacro)/xacro.py inorder $(find au_crustcrawler_base)/conf/our_crustcrawler.urdf.xacro"/-->

  <!--param name="robot_description_semantic"
      command="$(find xacro)/xacro.py inorder $(find crustcrawler)/config/crustcrawler.srdf"/-->

  <!-- OMPL Plugin for MoveIt! -->
  <arg name="planning_plugin" value="ompl_interface/OMPLPlanner" />

  <!-- The request adapters (plugins) used when planning with OMPL. 
       ORDER MATTERS -->
  
  

  <arg name="start_state_max_bounds_error" value="0.1" />

  <param name="planning_plugin" value="$(arg planning_plugin)" />
  
  <param name="start_state_max_bounds_error" value="$(arg start_state_max_bounds_error)" />

  <!--rosparam command="load" file="$(find crustcrawler)/config/kinematics.yaml"/>
  <rosparam command="load" file="$(find crustcrawler)/config/ompl_planning.yaml"/-->

  <rosparam command="load" file="$(find baxter_moveit_config)/config/kinematics.yaml"/>
  <rosparam command="load" file="$(find baxter_moveit_config)/config/ompl_planning.yaml"/>
  

  <node name="move_group" pkg="collision_free_traj" type="tests_sampling" output="screen" launch-prefix="gdb -ex run --args">
  <!--node name="in_and_out" pkg="collision_free_traj" type="tests_moveit" output="screen"-->
    <param name="planning_plugin" value="ompl_interface/OMPLPlanner"/>
    <param name="capabilities" value="move_group/MoveGroupCartesianPathService
				      move_group/MoveGroupExecuteService
				      move_group/MoveGroupKinematicsService
				      move_group/MoveGroupMoveAction
				      move_group/MoveGroupPlanService
				      move_group/MoveGroupQueryPlannersService
				      move_group/MoveGroupStateValidationService
				      move_group/MoveGroupGetPlanningSceneService
				      " />
   <param name="request_adapters" value="default_planner_request_adapters/AddTimeParameterization
				       default_planner_request_adapters/FixWorkspaceBounds
				       default_planner_request_adapters/FixStartStateBounds
				       default_planner_request_adapters/FixStartStateCollision
				       default_planner_request_adapters/FixStartStatePathConstraints" />
   <rosparam command="load" file="$(find collision_free_traj)/launch/work_space.yml"/>
  </node>
</launch>
